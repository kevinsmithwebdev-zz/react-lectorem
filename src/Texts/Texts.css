.Texts .block {
  display: flex;
  padding: 10px;
}

.Texts .block.even {
  background-color: #eee;
  border-radius: 2px;
}

.Texts .below {
  flex-direction: column;
  justify-content: start;
}

.Texts .beside {
  flex-direction: row;
}
