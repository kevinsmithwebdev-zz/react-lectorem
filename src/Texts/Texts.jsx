import React from 'react';
import TextBlock from './TextBlock/TextBlock';
import './Texts.css';
import TranslationBlock from './TranslationBlock/TranslationBlock';

const spreadParagraphsIntoSentenceBlocks = paragraphs => {
  const newBlocks = [];
  paragraphs.forEach(paragraph => {
    paragraph.sentences.forEach((sentence, idx) => {
      const newBlock = { sentences: [sentence] };
      if (idx === 0 && paragraph.header) {
        newBlock.header = paragraph.header;
      }
      newBlocks.push(newBlock);
    });
  });

  return newBlocks;
};

export const getClassName = (isTranslationBelow, idx) =>
  [
    'block',
    isTranslationBelow ? 'below' : 'beside',
    idx % 2 ? '' : 'even',
  ].join(' ');

const Texts = ({
  paragraphs, readTime, showModal, configuration,
}) => {
  if (!paragraphs) {
    return null;
  }

  const blocks = configuration.isBySentence
    ? spreadParagraphsIntoSentenceBlocks(paragraphs)
    : paragraphs;

  return (
    <div className='Texts'>
      {
        blocks.map((block, idx) => {
          const blockClassName = getClassName(configuration.isTranslationBelow, idx);
          return (
            <div key>
              <div className='header'>
                {
                  !!block.header?.imagePath && (
                    <div>
                      <img src={ block.header?.imagePath } alt={ `header ${idx}` } width='150px' />
                    </div>
                  )
                }
              </div>
              <div className={ blockClassName } key={ idx.toString() }>
                <TextBlock
                  configuration={ configuration }
                  block={ block }
                  idx={ idx }
                  readTime={ readTime }
                  showModal={ showModal }
                />
                <TranslationBlock
                  configuration={ configuration }
                  block={ block }
                />
              </div>
            </div>
          );
        })
      }
    </div>
  );
};

export default Texts;
