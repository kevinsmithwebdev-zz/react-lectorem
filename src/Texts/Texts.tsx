import React from 'react';
import TextBlock from './TextBlock/TextBlock';
import './Texts.css';
import TranslationBlock from './TranslationBlock/TranslationBlock';
import { ParagraphInterface, ConfigurationInterface } from '../interfaces/index';

export interface TextsInterface {
  paragraphs: ParagraphInterface[],
  readTime: number,
  showModal: Function,
  configuration: ConfigurationInterface,
}

export const spreadParagraphsIntoSentenceBlocks = (paragraphs: ParagraphInterface[]) => {
  const newBlocks: ParagraphInterface[] = [];
  paragraphs.forEach(paragraph => {
    paragraph.sentences.forEach((sentence, idx) => {
      const newBlock: ParagraphInterface = { sentences: [sentence] };
      if (idx === 0 && paragraph.header) {
        newBlock.header = paragraph.header;
      }
      newBlocks.push(newBlock);
    });
  });

  return newBlocks;
};

export const getClassName = (isTranslationBelow: boolean, idx: number) =>
  [
    'block',
    isTranslationBelow ? 'below' : 'beside',
    idx % 2 ? '' : 'even',
  ].join(' ');

const Texts: React.FC<TextsInterface> = ({
  paragraphs, readTime, showModal, configuration,
}) => {
  if (!paragraphs) {
    return null;
  }

  const blocks = configuration.isBySentence
    ? spreadParagraphsIntoSentenceBlocks(paragraphs)
    : paragraphs;

  return (
    <div className='Texts'>
      {
        blocks.map((block, idx) => {
          const blockClassName = getClassName(configuration.isTranslationBelow, idx);
          return (
            <div key={ idx.toString() }>
              <div className='header'>
                {
                  !!block.header?.imagePath && (
                    <div>
                      <img src={ block.header?.imagePath } alt={ `header ${idx}` } width='150px' />
                    </div>
                  )
                }
              </div>
              <div className={ blockClassName } key={ idx.toString() }>
                <TextBlock
                  configuration={ configuration }
                  block={ block }
                  idx={ idx }
                  readTime={ readTime }
                  showModal={ showModal }
                />
                <TranslationBlock
                  configuration={ configuration }
                  block={ block }
                />
              </div>
            </div>
          );
        })
      }
    </div>
  );
};

export default Texts;
