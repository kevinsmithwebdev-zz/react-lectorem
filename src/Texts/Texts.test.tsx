import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Texts, { TextsInterface, getClassName, spreadParagraphsIntoSentenceBlocks } from './Texts';
import storyData from '../fixtureData/story.data';
import paragraphsAsSentencesData from '../fixtureData/paragraphsAsSentences.data';

describe('Texts', () => {
  const { paragraphs } = storyData;
  describe('the main component render', () => {
    const readTime = 123.45;
    const showModal = () => {};

    const renderer = ShallowRenderer.createRenderer();

    describe('with no data, !isBySentence', () => {
      const props = {} as TextsInterface;
      it('should render and match snapshot', () => {
        renderer.render(<Texts {...props} />);
        expect(renderer.getRenderOutput()).toBeNull();
      });
    });

    describe('with data, !isBySentence', () => {
      const configuration = {
        isTranslationShown: false,
        isTranslationBelow: false,
        isBySentence: false,
      };
      const props = {
        paragraphs, readTime, showModal, configuration,
      };
      it('should render and match snapshot', () => {
        renderer.render(<Texts {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });

    describe('with data, isBySentence', () => {
      const configuration = {
        isTranslationShown: false,
        isTranslationBelow: false,
        isBySentence: true,
      };
      const props = {
        paragraphs, readTime, showModal, configuration,
      };
      it('should render and match snapshot', () => {
        renderer.render(<Texts {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });

  describe('spreadParagraphsIntoSentenceBlocks', () => {
    const actualReturn = spreadParagraphsIntoSentenceBlocks(paragraphs);
    it('should return the correct array', () => {
      expect(actualReturn).toStrictEqual(paragraphsAsSentencesData);
    });
  });

  describe('getClassName', () => {
    describe('when isTranslationBelow', () => {
      const isTranslationBelow = true;
      describe('when idx is odd', () => {
        const idx = 1;
        const expectedReturn = 'block below ';
        const actualReturn = getClassName(isTranslationBelow, idx);
        it('should return the correct string', () => {
          expect(actualReturn).toBe(expectedReturn);
        });
      });

      describe('when idx is even', () => {
        const idx = 2;
        const expectedReturn = 'block below even';
        const actualReturn = getClassName(isTranslationBelow, idx);
        it('should return the correct string', () => {
          expect(actualReturn).toBe(expectedReturn);
        });
      });
    });
  });

  describe('when !isTranslationBelow', () => {
    const isTranslationBelow = false;
    describe('when idx is odd', () => {
      const idx = 1;
      const expectedReturn = 'block beside ';
      const actualReturn = getClassName(isTranslationBelow, idx);
      it('should return the correct string', () => {
        expect(actualReturn).toBe(expectedReturn);
      });
    });

    describe('when idx is even', () => {
      const idx = 2;
      const expectedReturn = 'block beside even';
      const actualReturn = getClassName(isTranslationBelow, idx);
      it('should return the correct string', () => {
        expect(actualReturn).toBe(expectedReturn);
      });
    });
  });
});
