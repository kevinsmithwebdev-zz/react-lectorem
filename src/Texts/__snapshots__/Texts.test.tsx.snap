// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texts the main component render with data, !isBySentence should render and match snapshot 1`] = `
<div
  className="Texts"
>
  <div>
    <div
      className="header"
    >
      <div>
        <img
          alt="header 0"
          src="./frere.png"
          width="150px"
        />
      </div>
    </div>
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "header": Object {
              "imagePath": "./frere.png",
            },
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 5.511,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 4.424,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 6.533,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 5.51,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 7.683,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 6.618,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 8.79,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 7.68,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={0}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "header": Object {
              "imagePath": "./frere.png",
            },
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 5.511,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 4.424,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 6.533,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 5.51,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 7.683,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 6.618,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 8.79,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 7.68,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 10.857,
                    "start": 8.831,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 12.71,
                    "start": 11.025,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={1}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 10.857,
                    "start": 8.831,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 12.71,
                    "start": 11.025,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 13.706,
                    "start": 13.21,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 15.399,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 13.704,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 15.924,
                    "start": 15.39,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 17.572,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 15.923,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={2}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 13.706,
                    "start": 13.21,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 15.399,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 13.704,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 15.924,
                    "start": 15.39,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 17.572,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 15.923,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 19.319,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 17.57,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 21.632,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 19.784,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={3}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 19.319,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 17.57,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
              Object {
                "lects": Array [
                  Object {
                    "end": 21.632,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 19.784,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": false,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Texts the main component render with data, isBySentence should render and match snapshot 1`] = `
<div
  className="Texts"
>
  <div>
    <div
      className="header"
    >
      <div>
        <img
          alt="header 0"
          src="./frere.png"
          width="150px"
        />
      </div>
    </div>
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "header": Object {
              "imagePath": "./frere.png",
            },
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 5.511,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 4.424,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 6.533,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 5.51,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={0}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "header": Object {
              "imagePath": "./frere.png",
            },
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 5.511,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 4.424,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 6.533,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 5.51,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 7.683,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 6.618,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 8.79,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 7.68,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={1}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 7.683,
                    "explanation": "\\"Brother\\" in this sense means the title of a monk. \\"Frère\\" is also where we get our word \\"friar\\"",
                    "start": 6.618,
                    "text": "Frère",
                    "translation": "Brother",
                  },
                  Object {
                    "end": 8.79,
                    "explanation": "In the English version of the song this is rendered as \\"John\\" but a more accurate translation would be \\"James\\" or \\"Jacob\\", from the Latin \\"Iacobus\\".",
                    "start": 7.68,
                    "text": "Jacques",
                    "translation": "James",
                  },
                  Object {
                    "text": ".",
                  },
                ],
                "sentenceTranslation": "Brother James.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 10.857,
                    "start": 8.831,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={2}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 10.857,
                    "start": 8.831,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 12.71,
                    "start": 11.025,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={3}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 12.71,
                    "start": 11.025,
                    "text": "Dormez-vous?",
                    "translation": "Are you sleeping?",
                  },
                ],
                "sentenceTranslation": "Are you sleeping?",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 13.706,
                    "start": 13.21,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 15.399,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 13.704,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={4}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 13.706,
                    "start": 13.21,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 15.399,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 13.704,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 15.924,
                    "start": 15.39,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 17.572,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 15.923,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={5}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 15.924,
                    "start": 15.39,
                    "text": "Sonnez",
                    "translation": "Ring",
                  },
                  Object {
                    "end": 17.572,
                    "explanation": "These were the bells to wake the monks for the \\"matins\\", the first chants in the day, very early in the morning.",
                    "start": 15.923,
                    "text": "les matines",
                    "translation": "the morning bells",
                  },
                  Object {
                    "text": "!",
                  },
                ],
                "sentenceTranslation": "Ring the morning bells!",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside even"
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 19.319,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 17.57,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={6}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 19.319,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 17.57,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="header"
    />
    <div
      className="block beside "
    >
      <TextBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 21.632,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 19.784,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
        idx={7}
        readTime={123.45}
        showModal={[Function]}
      />
      <TranslationBlock
        block={
          Object {
            "sentences": Array [
              Object {
                "lects": Array [
                  Object {
                    "end": 21.632,
                    "explanation": "An onomatopoeia for bells, like our \\"ding dong\\".",
                    "start": 19.784,
                    "text": "Ding, dang, dong.",
                  },
                ],
                "sentenceTranslation": "Ding, dang, dong.",
              },
            ],
          }
        }
        configuration={
          Object {
            "isBySentence": true,
            "isTranslationBelow": false,
            "shouldShowTranslation": false,
          }
        }
      />
    </div>
  </div>
</div>
`;
